internal/velero/serverstatusrequest.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2020 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package velero
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/clock"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/buildinfo"
+	"github.com/vmware-tanzu/velero/pkg/plugin/framework"
+)
+
+type ServerStatus struct {
+	PluginRegistry PluginLister
+	Clock          clock.Clock
+}
+
+// PatchStatusProcessed patches status fields, including loading the plugin info, and updates
+// the ServerStatusRequest.Status.Phase to ServerStatusRequestPhaseProcessed.
+func (s *ServerStatus) PatchStatusProcessed(kbClient client.Client, req *velerov1api.ServerStatusRequest, ctx context.Context) error {
```

Comment:
We also have https://github.com/vmware-tanzu/velero/blob/main/pkg/client/dynamic.go#L78, which is generic regardless of fields, similar to `PatchHelper`.

I think I'd prefer going the generic route, seeing as 1) we have precedence in our dynamic client, though it'll eventually go away 2) we have an example in CAPI's and 3) we already have 2 examples in our own code in the BSL and ServerStatusRequest controllers, which is starting to be repetitive, even if they're not the same fields.

I think we're going to have something to gain from making a generic patching solution because a quick search of the code shows several controllers are doing patches:

```shell
Nolans-MBP in /Users/nrb/go/src/github.com/vmware-tanzu/velero/pkg (git) main
% ag patch -l --ignore apis --ignore generated --ignore backup/testdata
cmd/cli/plugin/remove.go
cmd/cli/plugin/add.go
serverstatusrequest/process.go
test/comparisons.go
test/fake_namespace.go
test/fake_dynamic.go
discovery/helper_test.go
controller/backupstoragelocation_controller.go
controller/backup_deletion_controller.go
controller/backupstoragelocation_controller_test.go
controller/restore_controller.go
controller/backup_sync_controller.go
controller/schedule_controller.go
controller/pod_volume_restore_controller.go
controller/download_request_controller.go
controller/backup_deletion_controller_test.go
controller/restic_repository_controller.go
controller/schedule_controller_test.go
controller/pod_volume_backup_controller.go
controller/restore_controller_test.go
controller/backup_controller.go
backup/backup.go
restore/merge_service_account_test.go
restore/merge_service_account.go
restore/restore.go
client/dynamic.go
```


That being said, I also don't know if it's worth holding up this PR for - it could go in on the next controller, too, or as its own PR that revisits the controllers.

Reply:
I'm ok with this direction.

I would prefer also not to hold up this PR for changing things to a generic helper. I've been thinking thru this as I work on the migration for the `downloadrequest`, so how about this: I'll add a generic helper with that change, then open a separate PR to change this and the other resources/controllers.